### PR TITLE
misc(scripts): github issue url scraper

### PR DESCRIPTION
**Pull request by connorjclark**
I wanted all the URLs mentioned in https://github.com/GoogleChrome/lighthouse/issues/6512, so I made this.

https://gist.github.com/connorjclark/d02c064ad3818a50b29498b4b3a48787

### CODE REVIEW

File: lighthouse-core/scripts/issue-url-scraper.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  const bars = new Array(Math.round(i * 40 / total)).fill('▄').join('').padEnd(40);
+  return `${i + 1} / ${total} [${bars}]`;
+}
+
+/**
+ * @param {string} qs
+ */
+async function query(qs) {
+  return await graphql(qs, {
+    owner: 'GooglChrome',
```
Comment (adamraine) [inline lighthouse-core/scripts/issue-url-scraper.js:42]: Is this a typo?

File: lighthouse-core/scripts/issue-url-scraper.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+ * @param {string} comment
+ */
+function parseCommentForUrl(comment) {
+  // https://stackoverflow.com/a/29288898
+  // eslint-disable-next-line max-len
+  const matches = /(?:(?:https?|file):\/\/|www\.)(?:\([-A-Z0-9+&@#/%=~_|$?!:,.]*\)|[-A-Z0-9+&@#/%=~_|$?!:,.])*(?:\([-A-Z0-9+&@#/%=~_|$?!:,.]*\)|[A-Z0-9+&@#/%=~_|$])/igm.exec(comment);
+  if (!matches) return null;
+
+  try {
+    const url = new URL(matches[0]);
```
Comment (adamraine) [inline lighthouse-core/scripts/issue-url-scraper.js:82]: If a comment links to a GH page before the failing URL, this function would return null right? Might be worthwhile to look at every match.

File: lighthouse-core/scripts/issue-url-scraper.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/scripts/issue-url-scraper.js:42]: haha, yup. worked anyway cuz this property isn't needed for auth.

File: lighthouse-core/scripts/issue-url-scraper.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/scripts/issue-url-scraper.js:82]: yeah I'll add that.

**Comment by brendankenny** [comment]
As more of a github utility than anything Lighthouse-specific, if it's not in an active workflow it might be better somewhere else?

Could still link in
https://github.com/GoogleChrome/lighthouse/blob/master/docs/hacking-tips.md or
https://github.com/GoogleChrome/lighthouse/blob/master/CONTRIBUTING.md if it seems helpful

**Comment by connorjclark** [comment]
> if it's not in an active workflow

I could see this being useful for any of our automated "close this and link to something else" issues.

Besides that, is saving a devdep worth the extra hops it'd take someone to use this / make & share improvements on? If just one other person uses it, I'd say no. If that's your main concern, I could move this into a folder and make a package.json, with directions that you need to be install it to work.

**Comment by brendankenny** [comment]
well it's 9 devDeps :)

I guess my reaction is what's different here than the thousands of github api scripts we've probably all accumulated over the years to do everything from scraping issues to calculating type coverage stats per commit, etc? I have several untracked directories in my local lighthouse checkout filled with scripts of varying levels of usefulness to others. "One other person might use one of them" doesn't seem like a helpful threshold for what I should open PRs for.

I don't know if there's an easily definable good threshold, but something like "in an automated or active workflow, otherwise go in docs or a different repo" was my off the cuff suggestion.

File: lighthouse-core/scripts/issue-url-scraper.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/scripts/issue-url-scraper.js]: this added 84 more urls

**Comment by connorjclark** [comment]
TODO: filter `https://user-images.githubusercontent.com/`, local ips. do basic `curl` check to see if url still valid.

maybe i'll make a npm package for this.

**Pull request closed without merging**